### DISCUSSION TITLE
Is there a way to have an Alpine listener on Livewire Components updates ?

### DISCUSSION THREAD

**Discussion by joffreypersia** [livewire/livewire#6932] [1. Help]
Hi, 

I am building a Livewire table and included Alpine inside. 
For some custom functionalities, I have Alpine that modifies the design, but when the user clicks on the pagination or changes the elements per page, only the `tbody` changes (the rows). 

Is it possible to have a listener like $watch but for Livewire morphs / changes so I can reset my style at this moment (trigger some 
functions) ? 

It could be great to have a magic property for it, since Alpine and Livewire are bounded now. 

Thank you

**Comment by joffreypersia** [comment]
Up please :)

**Comment by PhiloNL** [comment]
You can use the Livewire hook for this :) You would probably need to hook into `morph.updating`. 
See the documentation for more information:  https://livewire.laravel.com/docs/javascript#dom-morph-hooks

**Reply by joffreypersia** [reply]
Thank you for being so helpful, however, I couldn't make it fully work. 

I am not a very advanced coder, and I tried several things. 

If I just write in my x-data
```js
init() {
  $nextTick(() => {
    this.setTableHeight();
    Livewire.hook('morph.updated', ({ el, component }) => {
        console.log('Component has been updated!');
        this.setTableHeight();
     });
  });
  setTableHeight() {
     // my code
  }
}
```

It is trigger multiple times (for each DOM node inside my Livewire Component)

Isn't a way to just do something like ?
```js
@this.on('updated', () => { 
    this.setTableHeight
})
``` 

I found a workaround but I feel it is overkill and not very esthetic. 

```js
 init() {
     $nextTick(() => {
        this.setTableHeight();
        let isUpdated = false;

        Livewire.hook('morph.updated', ({ el, component }) => {
            if (!isUpdated) {
                isUpdated = true;
                setTimeout(() => {
                    console.log('Component has been updated!');

                    // Call your setTableHeight function or any other function you want to execute
                    this.setTableHeight();

                    // Reset the flag
                    isUpdated = false;
                }, 10);
            }
        });
    });
},
```

**Reply by joffreypersia** [reply]
Up ? :/
